services/autoscaling/src/simcore_service_autoscaling/utils/auto_scaling_core.py

```diff
@@ -159,24 +159,34 @@ def get_machine_buffer_type(
 
 DrainedNodes = list[AssociatedInstance]
 BufferDrainedNodes = list[AssociatedInstance]
+TerminatingNodes = list[AssociatedInstance]
 
 
 def sort_drained_nodes(
     app_settings: ApplicationSettings,
     all_drained_nodes: list[AssociatedInstance],
     available_ec2_types: list[EC2InstanceType],
-) -> tuple[DrainedNodes, BufferDrainedNodes]:
+) -> tuple[DrainedNodes, BufferDrainedNodes, TerminatingNodes]:
     assert app_settings.AUTOSCALING_EC2_INSTANCES  # nosec
+    # first sort out the drained nodes that started termination
+    terminating_nodes = [
```

Comment:
Q: why not use sets instead of lists. IMO sets are more suitable for the operations you are doing e.g.
```python
terminating_nodes: set
all_drained_nodes: set

remaining_nodes = all_drained_nodes - terminating_nodes
buffer_drained_nodes = remaining_drained_nodes - buffer_drained_nodes
```

Reply:
1. you need to be hashable to use sets, not always feasible
2. the sets might change the ordering of the stuff I am listing, which I do not necessarily want

but that is a good input.